### ISSUE TITLE
Docs: Remove redundant array literal in `defineConfig()` examples

### ISSUE THREAD

**Issue by kirkwaiblinger** [eslint/eslint#20736]
### Docs page(s)

https://eslint.org/docs/latest/use/configure/configuration-files
probably many more.

### What documentation issue do you want to solve?

Throughout the eslint site, the examples all seem to be structured as 
```js
export default defineConfig([
   configObject1,
   configObject2,
]);
```

but this is unnecessary; `defineConfig()` intentionally accepts a rest parameter, so that configs can be written directly as arguments:
```js
export default defineConfig(
   configObject1,
   configObject2,
);
```

I propose that all examples with `defineConfig([/* contents */])` be converted to `defineConfig(/* contents */)`. 

### What do you think is the correct solution?

Variadic input is an intentional feature of `defineConfig()` ([mentioned on RFC](https://github.com/eslint/rfcs/tree/main/designs/2024-config-extends#introduce-a-defineconfig-function)), so it's strange not to use it, and a bit clunky.

For intellisense/typing purposes, VSCode/TS does a better job at reporting type errors on objects specified as individual args than as part of an array literal (since it can inspect what's wrong inside each object, rather than just the array as a whole).

This also matches the documented usage of `defineConfig()`'s predecessor `tseslint.config()` better, and therefore would align better with both the historical and current typescript-eslint docs, which have always recommended the `defineConfig(object1, object2)` approach. 

### Participation

- [x] I am willing to submit a pull request for this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by lumirlumir** [comment]
I'm aware that on the typescript-eslint side, the recommended default configuration suggests using the rest-parameter form and it is widely used instead of passing an array.

https://typescript-eslint.io/getting-started/#step-2-configuration

However, one concern is that the 'pass an array' form is still used widely across the ESLint documentation, language plugins, the code explorer, the rewrite project example, and even some ESLint community projects. 

So, making it consistent would require significant effort, and I'm still not convinced how much benefit the rest-parameter form would bring relative to the time and effort required.

However, if the rest-parameter form is preferred after further discussion, AI could take on some of the workload, and there may be different opinions, leaving it open for further feedbacks. @eslint/eslint-team

**Comment by JoshuaKGoldberg** [comment]
I'm in favor of inlining. Anything we can do to simplify the docs helps. Those `[]` characters are a small "paper cut" that makes the docs just a little bit heavier to read.

> time and effort required

This could be a great `good first issue` kind of task! I don't think it'd be that much work: "just" find all instances of `defineConfig([` (around 119 at last count), remove the `[`, and remove the matching `]`. 

> the 'pass an array' form is still used widely across

Similarly, I think it'd be good to push the community to adopt. It's similarly good `good first issue` fodder for them.

**Comment by slaiff** [comment]
Hi, 

If this is still desired to be done, I would be happy to give it a shot. 

I'll work on it now and over the weekend.

**Comment by kecrily** [comment]
I agree with JoshuaKGoldberg. It’s not that complicated, and it really is a good first issue.

**Comment by lumirlumir** [comment]
> For intellisense/typing purposes, VSCode/TS does a better job at reporting type errors on objects specified as individual args than as part of an array literal (since it can inspect what's wrong inside each object, rather than just the array as a whole).

@kirkwaiblinger If you have a chance, could you share some more detailed examples that could help strengthen the OP? I’d like to add it to the agenda, and it seems like showing more benefits of this proposal would be helpful here.

**Comment by kirkwaiblinger** [comment]
@lumirlumir I played around this and unfortunately I actually couldn't find a strong example... looks like TS (current version, at least) is actually smart enough to inspect the contents of an array literal 🙂.

**Comment by lumirlumir** [comment]
**TSC Summary**: This issue is seeking consensus on removing redundant array literals in `defineConfig()` examples, but there isn’t definitive consensus yet.

**TSC Question**: Should we accept it, or if not, is it better to keep the current existing patterns?
